client/components/IndividualChallenge.js

```diff
@@ -1,107 +1,85 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect } from 'react';
 import CodeMirror from 'react-codemirror';
 import { connect } from 'react-redux';
 import { putUserchallenge } from '../store/userchallenge';
 
-class _IndividualChallenge extends Component {
-  constructor() {
-    super();
-    this.state = {
-      html: '',
-      css: '',
-      js: '',
-    };
-    this.updateCodeHTML = this.updateCodeHTML.bind(this);
-    this.updateCodeCSS = this.updateCodeCSS.bind(this);
-    this.updateCodeJS = this.updateCodeJS.bind(this);
-    this.changeValue = this.changeValue.bind(this);
-    this.putValue = this.putValue.bind(this);
-  }
+const _IndividualChallenge = props => {
```

Comment:
I get why you like hooks and agree they require writing less code.  I'm happy to use them as well.  The one thing I don't like about NOT using a class here is that you have a functional component that has methods of its own.  Not sure that is best practice.  Your thoughts?

Reply:
@theo333 not sure what the concern is here. Are you saying that a functional component shouldn't have private functions?